src/main/java/apoc/export/cypher/formatter/CreateCypherFormatter.java

```diff
@@ -13,37 +21,379 @@
+
+    }
+
+    private void writeUnwindStart(ExportConfig exportConfig, PrintWriter out, AtomicInteger batchCount) {
+        if (exportConfig.getFormat() == ExportFormat.CYPHER_SHELL
+                && exportConfig.getOptimizationType() == ExportConfig.OptimizationType.UNWIND_BATCH_PARAMS) {
+            if (batchCount.get() % exportConfig.getUnwindBatchSize() == 0) {
+                out.append(":param rows => [");
+            }
+        } else {
+            if (batchCount.get() % exportConfig.getUnwindBatchSize() == 0) {
+                if (batchCount.get() % exportConfig.getBatchSize() == 0) {
+                    out.append(exportConfig.getFormat().begin());
+                }
+                out.append("UNWIND [");
+            }
+        }
+    }
+
+    private void writeUnwindEnd(ExportConfig exportConfig, PrintWriter out, AtomicInteger batchCount, boolean isEnd) {
+        if (exportConfig.getFormat() == ExportFormat.CYPHER_SHELL
+                && exportConfig.getOptimizationType() == ExportConfig.OptimizationType.UNWIND_BATCH_PARAMS) {
+            if (isEnd || batchCount.get() % exportConfig.getUnwindBatchSize() == 0) {
+                out.append("]");
+                out.append(StringUtils.LF);
+                out.append(exportConfig.getFormat().begin());
+                out.append("UNWIND $rows AS row");
+            } else {
+                out.append(", ");
+            }
+        } else {
+            out.append("] as row");
+        }
+    }
+
+    private String getUniqueConstraintLabelsString(Node node, Map<String, Set<String>> uniqueConstraints) {
+        return CypherFormatterUtils.quote(String.join(":", getUniqueConstrainedLabels(node, uniqueConstraints)));
+    }
+
+    private Set<String> getUniqueConstrainedLabels(Node node, Map<String, Set<String>> uniqueConstraints) {
+        Set<String> constrainedLabels = uniqueConstraints.entrySet().stream()
+                .filter(e -> node.hasLabel(Label.label(e.getKey())) && e.getValue().stream().anyMatch(k -> node.hasProperty(k)))
+                .map(e -> e.getKey())
+                .collect(Collectors.toSet());
+
+        if (constrainedLabels.isEmpty()) {
+            constrainedLabels.add(CypherFormatterUtils.UNIQUE_ID_LABEL);
+        }
+        return constrainedLabels;
+    }
+
+    private Set<String> getLabels(Node node) {
+        Set<String> labels = StreamSupport.stream(node.getLabels().spliterator(), false)
+                .map(Label::name)
+                .collect(Collectors.toSet());
+        if (labels.isEmpty()) {
+            labels.add(CypherFormatterUtils.UNIQUE_ID_LABEL);
+        }
+        return labels;
+    }
+
+    private void writeAsciiNode(Node node, PrintWriter out, String key, Map.Entry<Set<String>, Set<String>> entry, Map<String, Set<String>> uniqueConstraints) {
+        Set<String> uniqueConstraintsLabels = getUniqueConstrainedLabels(node, uniqueConstraints);
+        out.append(StringUtils.LF);
+        out.append("MATCH ");
```

Comment:
confusing that "node" implies MATCH I would have thought that's a separate concern from the pattern